src/librbd/AioImageRequest.cc

```diff
@@ -447,7 +447,7 @@ void AioImageFlush::send_request() {
                   !m_image_ctx.journal->is_journal_replaying());
   }
 
-  m_aio_comp->set_request_count(cct, journaling ? 2 : 1);
```

Comment:
You will want this original line -- perhaps just move it inside the if and inside the else block so that you can set it to 2 and 1, respectively